packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -1,41 +1,60 @@
 import React, { KeyboardEventHandler, useEffect, useState } from 'react'
-import {
-  Box,
-  List,
-  Paper,
-  Typography,
-  FormControl,
-  InputAdornment,
-  IconButton,
-} from '@mui/material'
+import { Box, List, Paper, Typography, IconButton } from '@mui/material'
 import { useTheme } from '@mui/material'
 import { ColorPartial } from '@mui/material/styles/createPalette'
-import { OutlinedInput } from '@mui/material'
 import { ArrowUpward } from '@mui/icons-material'
 import SynapseChatInteraction from './SynapseChatInteraction'
 import { SkeletonParagraph } from '../Skeleton'
-import { useCreateAgentSession } from 'src/synapse-queries/chat/useChat'
+import {
+  useCreateAgentSession,
+  useGetAgentChatWithAsyncStatus,
+} from '../../synapse-queries/chat/useChat'
 import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+import { TextField } from '@mui/material'
 
 export type SynapseChatProps = {
-  initialMessage: string
+  initialMessage?: string //optional initial message
   agentId?: string // if provided, use this agent
 }
 
+type ChatInteraction = {
+  userMessage: string
+  chatResponseText?: string
+}
 export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
   initialMessage,
   agentId,
 }) => {
-  // Create a new Agent Session on mount, add ability to initialize this component with a starting message from the user.
-  const [agentSessionId, setAgentSessionId] = useState('')
-  const { mutate: createAgentSession } = useCreateAgentSession({
-    onSuccess: newAgentSession => {
-      setAgentSessionId(newAgentSession.sessionId)
-    },
-  })
+  const { data: agentSession, mutate: createAgentSession } =
+    useCreateAgentSession()
   const theme = useTheme()
-  const [messages, setMessages] = useState([initialMessage])
+  // When both the current message and current response are available, add a new ChatInteraction to the array
+  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
   const [currentMessage, setCurrentMessage] = useState('')
+  const [currentResponse, setCurrentResponse] = useState('')
+  // Keep track of the text that the user is currently typing into the textfield
+  const [userChatTextfieldValue, setUserChatTextfieldValue] = useState('')
+
+  const { mutate: createAgentChatInteraction } = useGetAgentChatWithAsyncStatus(
+    {
+      onSuccess: data => {
+        // whenever the response is returned, set the last interaction response text
+        setCurrentResponse(data.responseText)
+      },
+    },
+  )
+
+  useEffect(() => {
+    // when we have both a message and response, add a new interaction and reset
+    if (currentMessage && currentResponse) {
+      setInteractions([
+        ...interactions,
+        { userMessage: currentMessage, chatResponseText: currentResponse },
+      ])
+      setCurrentResponse('')
+      setCurrentMessage('')
+    }
+  }, [currentMessage, currentResponse])
```

Comment:
Why not just do this in the `onSuccess` callback passed to `useGetAgentChatWithAsyncStatus`? And maybe just disable the TextField while the mutation is pending?